packages/zowe-explorer/src/commands/ZoweCommandProvider.ts

```diff
@@ -19,18 +19,225 @@ import { Constants } from "../configuration/Constants";
+
+                                const diagInfo: string = error.details.additionalDetails;
+                                if (diagInfo?.length > 0) {
+                                    formattedError += imperative.TextUtils.chalk.bold.yellow(`\n${vscode.l10n.t("Diagnostic Information")}\n`);
+                                    formattedError += diagInfo;
+                                }
+
+                                return formattedError;
+                            }
+                            return error.message;
+                        }
+                    },
+                    {
+                        message: vscode.l10n.t({
+                            message: "Welcome to the integrated terminal for: {0}",
+                            args: [this.terminalName],
+                            comment: ["Terminal Name"],
+                        }),
+                        history: [...this.history.getSearchHistory()].reverse(),
+                        startup: command,
+                        formatCommandLine: (cmd: string) => this.formatCommandLine(cmd, profile),
+                    }
+                );
+                this.terminal = vscode.window.createTerminal({ name: `(${profile.name}) ${this.terminalName}`, pty: this.pseudoTerminal });
+                this.terminal.show();
+            } else {
+                this.outputChannel ??= Gui.createOutputChannel(this.terminalName);
+                this.outputChannel.appendLine(this.formatCommandLine(command, profile));
+                const response = await Gui.withProgress(
+                    {
+                        location: vscode.ProgressLocation.Notification,
+                        title: this.dialogs.commandSubmitted,
+                    },
+                    () => {
+                        return this.runCommand(profile, command);
+                    }
+                );
+                this.outputChannel.appendLine(response);
+                this.outputChannel.show(true);
+                this.history.addSearchHistory(command);
+            }
+        } catch (error) {
+            await AuthUtils.errorHandling(error, { apiType: ZoweExplorerApiType.Command, profile });
+        }
+    }
+
+    public async selectNodeProfile(cmdTree: Definitions.Trees): Promise<imperative.IProfileLoaded> {
+        ZoweLogger.trace("ZoweCommandProvider.selectNodeProfile called.");
+
+        const profileNamesList = ProfileManagement.getRegisteredProfileNameList(cmdTree);
+        if (profileNamesList.length > 0) {
+            const quickPickOptions: vscode.QuickPickOptions = {
+                placeHolder: this.dialogs.selectProfile,
+                ignoreFocusOut: true,
+                canPickMany: false,
+            };
+            const sesName = await Gui.showQuickPick(profileNamesList, quickPickOptions);
+            if (sesName === undefined) {
+                ZoweLogger.info(this.operationCancelled);
+                return;
```

Comment:
Is this log message needed? If so, can we make it more specific to point out exactly what operation was cancelled? From the perspective of a user reading the output channel, all that would be printed is `[INFO] Operation cancelled`.